TypeCobol.Test/Parser/Preprocessor/TestReplaceDirective.cs

```diff
@@ -9,10 +9,12 @@ public static void CheckReplaceSingle()
             PreprocessorUtils.CheckWithReplaceResultFile(result, testName);
         }
 
-        public static void CheckReplacePartial()
+        public static void CheckReplacePartial(bool cobol)
         {
-            string testName = "PgmReplacePartial";
+            string testName = cobol ? "PgmReplacePartial" : "PgmReplacePartialTC";
+            PreprocessorUtils.ReplaceProject.CompilationOptions.IsCobolLanguage = cobol;
             string result = PreprocessorUtils.ProcessReplaceDirectives(testName);
+            PreprocessorUtils.ReplaceProject.CompilationOptions.IsCobolLanguage = false;
```

Comment:
Can you add a param in `ProcessReplaceDirectives` instead of manipulating the options from outside ?

Reply:
In fact PreprocessorUtils.ReplaceProject.CompilationOptions is a static shared resource, it is the resposability of the caller to do that.

Reply:
Yes I remember this is some sort of hack, I had to deal with it in #1890. Ok.

Reply:
About the previous remark, you could use the same `testName` for the `ProcessReplaceDirectives` call thus using a single input source file, and alter the name for the result comparison depending on the `cobol` parameter. Would it work ?